File: 2024-2/0684/unidad04.c - Statement 01/SoftPubPersonaBO/src/pe/edu/pucp/softpub/persona/bo/SeccionBO.java
package pe.edu.pucp.softpub.persona.bo;

import pe.edu.pucp.softpub.persona.dao.SeccionDAO;
import pe.edu.pucp.softpub.persona.daoImp.SeccionDAOImpl;
import pe.edu.pucp.softpub.persona.model.Seccion;

public class SeccionBO {
    
    private SeccionDAO seccionDAO;
    
    public SeccionBO(){
        this.seccionDAO = new SeccionDAOImpl();
    }
    
    public Integer insertar(String idSeccion, String nombre_seccion, String departamento){
        Seccion seccion = new Seccion(idSeccion, nombre_seccion, departamento);
        return seccionDAO.insertar(seccion);
    }
}

File: 2024-2/0682/unidad06.c_Web_03/SofPub/SoftPubPersonaBO/SeccionBO.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using SoftPubPersonaDA.DAO;
using SoftPubPersonaDA.DAOImpl;
using SoftPubPersonaModel;

namespace SoftPubPersonaBO
{
    public class SeccionBO
    {
        private SeccionDAO seccionDAO;

        public SeccionBO() 
        { 
            this.seccionDAO = new SeccionDAOImpl();
        }

        public int insertar(string idSeccion, string nombre_seccion, string departamento)
        {
            Seccion seccion = new Seccion(idSeccion, nombre_seccion, departamento);
            return this.seccionDAO.insertar(seccion);
        }

        public int modificar(string idSeccion, string nombre_seccion, string departamento)
        {
            Seccion seccion = new Seccion(idSeccion, nombre_seccion, departamento);
            return this.seccionDAO.modificar(seccion);
        }

        public int eliminar(string idSeccion)
        {
            Seccion seccion = new Seccion();
            seccion.IdSeccion = idSeccion;
            return this.seccionDAO.eliminar(seccion);
        }

        public BindingList<Object> listarTodos()
        {
            return this.seccionDAO.listarTodos();
        }
    }
}
